test/test_helper.rb

```diff
@@ -16,6 +16,17 @@ class ActionView::TestCase
 
 class ActiveSupport::TestCase
   include FactoryBot::Syntax::Methods
+
+  def stub_repo_cache!(&block)
+    git_repo_cache = "#{Rails.root}/test/tmp/git_repo_cache"
+    Git::RepoBase.
+      stubs(:repo_base_dir).
+      returns(git_repo_cache)
+
+    yield
+
+    FileUtils.rm_rf(git_repo_cache)
```

Comment:
Is it worth wrapping this in an `ensure`?

Reply:
Yes! Thank you :)